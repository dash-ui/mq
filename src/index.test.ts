import mq from './index'

const breakpoints = {
  // 0px
  phone: 'only screen and (min-width: 0em)',
  // 560px
  tablet: 'only screen and (min-width: 35em)',
  // 1280px
  desktop: 'only screen and (min-width: 80em)',
  'hi-dpi': `
     only screen and (-webkit-min-device-pixel-ratio: 1.5),
     only screen and (min--moz-device-pixel-ratio: 1.5),
     only screen and (-o-min-device-pixel-ratio: 1.5/1),
     only screen and (min-resolution: 144dpi),
     only screen and (min-resolution: 1.5dppx)
  `,
}

describe('mq()', () => {
  it('should return a media query string w/ one argument', () => {
    const breakpoint = mq(breakpoints)
    expect(breakpoint('phone')).toBe('@media only screen and (min-width: 0em)')
  })

  it('should return a media query string w/ one object argument', () => {
    const breakpoint = mq(breakpoints)
    expect(breakpoint({phone: true})).toBe(
      '@media only screen and (min-width: 0em)'
    )
  })

  it('should return a media query string w/ two object argument', () => {
    const breakpoint = mq(breakpoints)
    expect(
      breakpoint(
        {phone: true},
        ({blue}) => `color: ${blue};`
      )({blue: 'var(--blue)'})
    ).toMatchSnapshot('var(--blue)')

    expect(
      breakpoint({phone: true}, 'color: red;')({blue: 'var(--blue)'})
    ).toMatchSnapshot('red')

    expect(
      breakpoint({phone: true}, {color: 'green'})({blue: 'var(--blue)'})
    ).toMatchSnapshot('green')
  })

  it('should join multiple query object media queries into one', () => {
    const breakpoint = mq(breakpoints)
    expect(
      breakpoint(
        {phone: true, 'hi-dpi': true},
        ({blue}) => `color: ${blue};`
      )({blue: 'var(--blue)'})
    ).toMatchSnapshot('phone, hi-dpi')
  })

  it('should not create media queries for falsy values', () => {
    const breakpoint = mq(breakpoints)
    expect(
      breakpoint(
        {phone: true, 'hi-dpi': 0},
        ({blue}) => `color: ${blue};`
      )({blue: 'var(--blue)'})
    ).toMatchSnapshot('phone, hi-dpi')

    expect(
      breakpoint({phone: false, 'hi-dpi': 0}, ({blue}) => `color: ${blue};`)
    ).toMatchSnapshot('empty string')
  })

  it('should return a style getter when there is a second argument', () => {
    const breakpoint = mq(breakpoints)
    expect(
      breakpoint('phone', ({blue}) => `color: ${blue};`)({blue: 'var(--blue)'})
    ).toMatchSnapshot('var(--blue)')

    expect(
      breakpoint('phone', 'color: red;')({blue: 'var(--blue)'})
    ).toMatchSnapshot('red')

    expect(
      breakpoint('phone', {color: 'green'})({blue: 'var(--blue)'})
    ).toMatchSnapshot('green')
  })
})
