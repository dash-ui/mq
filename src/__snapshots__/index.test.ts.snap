// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mq() should join multiple query object media queries into one: phone, hi-dpi 1`] = `
"@media only screen and (min-width: 0em),
     only screen and (-webkit-min-device-pixel-ratio: 1.5),
     only screen and (min--moz-device-pixel-ratio: 1.5),
     only screen and (-o-min-device-pixel-ratio: 1.5/1),
     only screen and (min-resolution: 144dpi),
     only screen and (min-resolution: 1.5dppx)
  {color:var(--blue);}"
`;

exports[`mq() should not create media queries for falsy values: empty string 1`] = `""`;

exports[`mq() should not create media queries for falsy values: phone, hi-dpi 1`] = `"@media only screen and (min-width: 0em){color:var(--blue);}"`;

exports[`mq() should return a media query string w/ two object argument: green 1`] = `"@media only screen and (min-width: 0em){color:green;}"`;

exports[`mq() should return a media query string w/ two object argument: red 1`] = `"@media only screen and (min-width: 0em){color:red;}"`;

exports[`mq() should return a media query string w/ two object argument: var(--blue) 1`] = `"@media only screen and (min-width: 0em){color:var(--blue);}"`;

exports[`mq() should return a style getter when there is a second argument: green 1`] = `"@media only screen and (min-width: 0em){color:green;}"`;

exports[`mq() should return a style getter when there is a second argument: red 1`] = `"@media only screen and (min-width: 0em){color:red;}"`;

exports[`mq() should return a style getter when there is a second argument: var(--blue) 1`] = `"@media only screen and (min-width: 0em){color:var(--blue);}"`;
